test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/EnumFilterFunctionalTests.cs

```diff
@@ -515,28 +520,245 @@ public void ParseFilterEnumTypesWrongCast3()
+        {
+            // Arrange
+            int enumValue = 1;
+            string filterQuery = $"'{enumValue}' in Colors"; // "'1' in Colors"
+            string expectedLiteral = "'Red'";
+
+            IEdmEnumType colorType = this.GetIEdmType<IEdmEnumType>("NS.Color");
+            bool success = colorType.TryParse(enumValue, out IEdmEnumMember enumMember);
+
+            // Act
+            FilterClause filterQueryNode = ParseFilter(filterQuery, this.userModel, this.entityType, this.entitySet);
+            SingleValueNode expression = filterQueryNode.Expression;
+
+            // Assert
+            Assert.Equal("'1' in Colors", filterQuery);
+            Assert.True(success);
+            InNode inNode = expression.ShouldBeInNode();
+            ConstantNode leftNode = Assert.IsType<ConstantNode>(inNode.Left);
+            leftNode.ShouldBeEnumNode(colorType, enumMember.Name);
+
+            Assert.True(leftNode.TypeReference.IsEnum());
+            Assert.Equal(enumMember.Name, leftNode.Value.ToString());
+            Assert.Equal(expectedLiteral, leftNode.LiteralText);
+
+            CollectionPropertyAccessNode rightNode = Assert.IsType<CollectionPropertyAccessNode>(inNode.Right);
+            rightNode.ShouldBeCollectionPropertyAccessQueryNode(this.GetIEdmProperty("Colors"));
+            Assert.Equal(colorType, rightNode.ItemType.Definition);
+        }
+
+        [Fact]
+        public void ParseFilterWithInOperatorWithEnumsMemberIntegralValueWithoutSingleQuotes()
+        {
+            // Arrange
+            int enumValue = 3;
+            string filterQuery = $"{enumValue} in Colors"; // "3 in Colors"
+            string expectedLiteral = "'Blue'";
+
+            IEdmEnumType colorType = this.GetIEdmType<IEdmEnumType>("NS.Color");
+            bool success = colorType.TryParse(enumValue, out IEdmEnumMember enumMember);
+
+            // Act
+            FilterClause filterQueryNode = ParseFilter(filterQuery, this.userModel, this.entityType, this.entitySet);
+            SingleValueNode expression = filterQueryNode.Expression;
+
+            // Assert
+            Assert.Equal("3 in Colors", filterQuery);
+            Assert.True(success);
+            InNode inNode = expression.ShouldBeInNode();
+            ConstantNode leftNode = Assert.IsType<ConstantNode>(inNode.Left);
+            leftNode.ShouldBeEnumNode(colorType, enumMember.Name);
+
+            Assert.True(leftNode.TypeReference.IsEnum());
+            Assert.Equal(enumMember.Name, leftNode.Value.ToString());
+            Assert.Equal(expectedLiteral, leftNode.LiteralText);
+
+            CollectionPropertyAccessNode rightNode = Assert.IsType<CollectionPropertyAccessNode>(inNode.Right);
+            rightNode.ShouldBeCollectionPropertyAccessQueryNode(this.GetIEdmProperty("Colors"));
+            Assert.Equal(colorType, rightNode.ItemType.Definition);
+        }
+
+        [Fact]
+        public void ParseFilterWithInOperatorWithEnumsMemberDoubleIntegralValue()

```

Comment:
What do you mean by `DoubleIntegralValues`? How is this test different from this: https://github.com/OData/odata.net/pull/3023/files#diff-9f67b5b315ba1bb4ddbcb823f5c4183658c6a7f7a3b06d0da900b0ea7e21c7d6R614

Reply:
@habbes: I meant to test for `long` type.
I have removed this test as it is similar to https://github.com/OData/odata.net/pull/3023/files#diff-9f67b5b315ba1bb4ddbcb823f5c4183658c6a7f7a3b06d0da900b0ea7e21c7d6R614